### ISSUE TITLE
`createInterpolateElement()` triggers `jsx-a11y/anchor-has-content`

### ISSUE THREAD

**Issue by iandunn** [WordPress/gutenberg#21441]
### Problem

#20699 introduced `createInterpolateElement()`, but using it with anchor tags triggers the [`jsx-a11y/anchor-has-content`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md) rule. e.g.,

```js
createInterpolateElement(
	__( 'There was an error trying to render this content, please try another browser or device. If that doesn\'t work, please <a>contact us</a>.', 'wordcamporg' ),
	{
		a: <a href={ 'https://central.wordcamp.org/contact-us/' } />,
	}
);
```
```sh
> npm run lint:js source/utils/render-frontend.js

...
63:9   error  Anchors must have content and the content must be accessible by a screen reader  jsx-a11y/anchor-has-content
```

### Potential solutions

I'm not seeing any great solutions at first glance.

* Disable the linter rule
* Replace it with a custom rule that is aware of `createInterpolateElement()`
* Refactor `createInterpolateElement()` to avoid triggering the rule.

**Comment by iandunn** [comment]
In the meantime, one workaround is to do something like this:

`a: <a href={ 'https://central.wordcamp.org/contact-us/' } >#21441-gutenberg</a>`

That way the rule isn't triggered, and the reason for the seemingly unnecessary anchor text is self-documented.

**Comment by gziolo** [comment]
It’s definitely an edge case for the `jsx-a11y/anchor-has-content` rule. It’s something that is probably fine to disable when spotted in the code.

It would be also great to have a custom wrapping rule that is able to detect context and cover this specific case as an exception.

**Comment by rowasc** [comment]
@gziolo hi - I made a [first pass on this](https://github.com/WordPress/gutenberg/pull/50065/files) while trying to learn a bit more about Gutenberg. It's pretty basic but I think it should help with not having to manually ignore each usage of createInterpolateElement + anchors. 

It's a draft (e.g: missing docs) but I thought I'd let you know in case someone has strong opinions against doing this or the general approach.

**Comment by gziolo** [comment]
> It's a draft (e.g: missing docs) but I thought I'd let you know in case someone has strong opinions against doing this or the general approach.

Excellent, thank you so much for your contribution!
